integration_tests/src/main/python/parquet_write_test.py

```diff
@@ -567,3 +568,76 @@ def test_write_empty_data_single_writer(spark_tmp_path):
+           the `type` column.
+        3. Populates the partitioned table using the un-partitioned table as input.
+             1. If the partition mode is dynamic, all the partitions are populated at once.
+             2. If the partition mode is static, the partitions are populated explicitly,
+                and sequentially.
+        4. Returns all the rows from 2 of the 3 partitions in the new table.
+    """
+
+    def create_input_table(spark):
+        tmp_input = spark_tmp_table_factory.get()
+        spark.sql("CREATE TABLE " + tmp_input +
+                  " (make STRING, model STRING, year INT, type STRING, comment STRING)" +
+                  " STORED AS PARQUET")
+        spark.sql("INSERT INTO TABLE " + tmp_input + " VALUES " +
+                  "('Ford',   'F-150',       2020, 'ICE',      'Popular' ),"
+                  "('GMC',    'Sierra 1500', 1997, 'ICE',      'Older'),"
+                  "('Chevy',  'D-Max',       2015, 'ICE',      'Isuzu?' ),"
+                  "('Tesla',  'CyberTruck',  2025, 'Electric', 'Fictional'),"
+                  "('Rivian', 'R1T',         2022, 'Electric', 'Heavy'),"
+                  "('Jeep',   'Gladiator',   2024, 'Hybrid',   'Upcoming')")
+        return tmp_input
+
+    def populate_partitions_static(spark):
+        input_table = create_input_table(spark)
+        output_table = spark_tmp_table_factory.get()
+        spark.sql("CREATE TABLE " + output_table +
+                  " (make STRING, model STRING, year INT, comment STRING)"
+                  " PARTITIONED BY (type STRING) STORED AS PARQUET")
+        spark.sql("INSERT INTO TABLE " + output_table + " PARTITION (type='ICE')" +
+                  " SELECT make, model, year, comment FROM " + input_table +
+                  " WHERE type='Electric'")
+        spark.sql("INSERT OVERWRITE TABLE " + output_table + " PARTITION (type='ICE')" +
+                  " SELECT make, model, year, comment FROM " + input_table +
+                  " WHERE type='Hybrid'")
+        return spark.sql("SELECT * FROM " + output_table +
+                         " WHERE type = 'ELECTRIC' or type = 'Hybrid'")
+
+    def populate_partitions_dynamic(spark):
+        input_table = create_input_table(spark)
+        output_table = spark_tmp_table_factory.get()
+        spark.sql("CREATE TABLE " + output_table +
+                  " (make STRING, model STRING, year INT, comment STRING)"
+                  " PARTITIONED BY (type STRING) STORED AS PARQUET")
+        spark.sql("INSERT OVERWRITE TABLE " + output_table +
+                  " SELECT make, model, year, comment, type FROM " + input_table)
+        return spark.sql("SELECT * FROM " + output_table +
+                         " WHERE type = 'ELECTRIC' or type = 'Hybrid'")
+
+    return populate_partitions_static if mode == PartitionWriteMode.Static else populate_partitions_dynamic
+
+
+@allow_non_gpu("EqualTo,IsNotNull,Literal,Or")  # Accounts for partition predicate.
```

Comment:
I'm confused why this is here.  We don't normally need to exclude things like this, see the other partitioned writes in parquet_write_test.py.

Reply:
I should have made this clearer in the documentation: this is to prevent the test from mistakenly flagging the partition predicates.
The test writes 3 Hive partitions, and reads back 2:
```sql
SELECT * FROM foobar WHERE type = 'Electric' OR type = 'Hybrid' ;
```

This produces partition predicates like:
```scala
PartitionFilters: [((type#87 = Electric) OR (type#87 = Hybrid))], 
```
There is a "not-null" predicate as well. These run on the Hive Metastore, and can't be GPU accelerated.

The other tests don't seem to be specifying a partition predicate in the read; they're reading all the data.

Reply:
I'll see about using `assert_gpu_and_cpu_sql_writes_are_equal_collect()`. That should obviate the need to whitelist these operators.

Reply:
> These run on the Hive Metastore, and can't be GPU accelerated.

Then this indicates that there's a GPU node that should have overridden `gpuExpressions` but did not.  We have other examples of CPU expressions being used directly (e.g.: for DPP or predicate pushdown), but we don't have to specify an allow list for these in the tests.

Reply:
I've filed #7662. In the meantime, I've switched to using `assert_gpu_and_cpu_sql_writes_are_equal_collect()`.

One offshoot of this change is that the read function does not specify a partition predicate. I don't think we've lost test coverage, because the assert function uses `SELECT *`, which also selects the partition key.